### PR TITLE
User routes

### PR DESCRIPTION
**Pull request by thinknoack**

### CODE REVIEW

File: packages/openneuro-app/src/scripts/users/user-account-container.tsx [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+
+const UserAccountContainer: React.FC<AccountContainerProps> = ({ user }) => {
+
+  return (
+    <>
+        <div className="user-info">
+            <div className="user-account-card">{user.id}</div>
+        <div className="user-account-tab-links">ComponentsLinks to datasets, account, and notifications
+            <Link to={'account'}> Test </Link>
```
Comment (nellh) [inline packages/openneuro-app/src/scripts/users/user-account-container.tsx]:
```suggestion
            <Link to="account"> Test </Link>
```

File: packages/openneuro-app/src/scripts/users/user-query.tsx [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+ * Routing wrapper for USER query
+ *
+ * Expects to be a child of a react-router Route component with UserId
+ */
+const UserQuery = () => {
+    const { user } = useParams()
+    return (
+        <>
+            <UserQueryHook user={user} />
+        </>
```
Comment (nellh) [inline packages/openneuro-app/src/scripts/users/user-query.tsx]:
```suggestion
        <UserQueryHook user={user} />
```

File: packages/openneuro-app/src/scripts/routes.tsx [hunk 3]
```diff
@@ -33,6 +35,7 @@ const AppRoutes: React.VoidFunctionComponent = () => (
     <Route path="/import" element={<ImportDataset />} />
     <Route path="/metadata" element={<DatasetMetadata />} />
     <Route path="/public" element={<Navigate to="/search" replace />} />
+    <Route path="/account/:orchid/*" element={<UserQuery/>} />
```
Comment (nellh) [inline packages/openneuro-app/src/scripts/routes.tsx]:
```suggestion
    <Route path="/account/:orcid/*" element={<UserQuery/>} />
```

File: packages/openneuro-app/src/scripts/users/user-routes.tsx [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+import PropTypes from "prop-types"
+import { Route, Routes } from "react-router-dom"
+import UserAccountContainer from "./user-account-container"
+
+const DatasetRoutes = ({ user, error }) => {
+  useEffect(() => {
+    if (error) {
+      throw error
+    }
+  }, [user, error])
```
Comment (nellh) [inline packages/openneuro-app/src/scripts/users/user-routes.tsx]:
```suggestion
```
I don't think this is actually useful. The error will bubble up anyways.

File: packages/openneuro-app/src/scripts/users/user-routes.tsx [hunk 5]
```diff
@@ -0,0 +1,29 @@
+import React, { useEffect } from "react"
+import PropTypes from "prop-types"
+import { Route, Routes } from "react-router-dom"
+import UserAccountContainer from "./user-account-container"
+
+const DatasetRoutes = ({ user, error }) => {
```
Comment (nellh) [inline packages/openneuro-app/src/scripts/users/user-routes.tsx]:
```suggestion
export const UserRoutes = ({ user, error }) => {
```

File: packages/openneuro-app/src/scripts/users/user-routes.tsx [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+    </Routes>
+  )
+}
+
+DatasetRoutes.propTypes = {
+  dataset: PropTypes.object,
+  error: PropTypes.object,
+}
+
+export default DatasetRoutes
```
Comment (nellh) [inline packages/openneuro-app/src/scripts/users/user-routes.tsx]:
```suggestion
```
We have a lot of default exports but generally better to use named exports for components.

File: packages/openneuro-app/src/scripts/users/user-query.tsx [hunk 7]
```diff
@@ -0,0 +1,79 @@
+import React from 'react'
+import * as Sentry from '@sentry/react'
+import PropTypes from 'prop-types'
+import { useParams } from 'react-router-dom'
+import { Loading } from '@openneuro/components/loading'
+import UserRoutes from './user-routes'
```
Comment (nellh) [inline packages/openneuro-app/src/scripts/users/user-query.tsx]:
```suggestion
import { UserRoutes } from './user-routes'
```

**Review by nellh — COMMENTED** [review]
Looks like this is on the right track. Added some suggestions but it's mostly minor fixes.

File: packages/openneuro-app/src/scripts/users/user-query.tsx (hunk 7 above)
Comment (thinknoack) [reply packages/openneuro-app/src/scripts/users/user-query.tsx]: I've updated files I created to be named exports. Should I go ahead and do the same for ones I see like `<FourOFour />` etc.

**Comment by thinknoack** [comment]
> Looks like this is on the right track. Added some suggestions but it's mostly minor fixes.

ok, lint/ts issues have been resolved.

File: packages/openneuro-app/src/scripts/users/__tests__/user-account-view.tsx [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+    links: ["https://example.com", "https://example.org"],
+    github: "johndoe",
+  };
+
+describe('<UserAccountView />', () => {
+  it('should render the user details correctly', () => {
+    render(<UserAccountView user={baseUser} />);
+
+    // Check if user details are rendered
+    expect(screen.getByText('Name:')).toBeInTheDocument();
```
Comment (nellh) [inline packages/openneuro-app/src/scripts/users/__tests__/user-account-view.tsx]: I think checking for this by name is fine but generally looking for the dynamic data is going to be more important and this more tests than needed can be brittle.

File: packages/openneuro-app/src/scripts/users/__tests__/user-account-view.tsx [hunk 9]
```diff
@@ -0,0 +1,79 @@
...
+
+  it('should render links with EditableContent', () => {
+    render(<UserAccountView user={baseUser} />);
+
+    // Editable Links section
+    const linksSection = screen.getByText('Links');
+    expect(linksSection).toBeInTheDocument();
+
+    const linkItems = screen.getAllByRole('listitem');
+    expect(linkItems).toHaveLength(2); // initially 2 links
```
Comment (nellh) [inline packages/openneuro-app/src/scripts/users/__tests__/user-account-view.tsx]: Get by role here is great, this is a good example of a test that checks the functionality of the component.

File: packages/openneuro-app/src/scripts/users/__tests__/user-account-view.tsx
Comment (nellh) [inline packages/openneuro-app/src/scripts/users/__tests__/user-account-view.tsx:1]: By convention this should be user-account-view.spec.tsx. These files should probably all be .spec.tsx to match the rest of the project.

**Review by nellh — APPROVED** [review]
Approved with minor changes for the test filenames.

File: packages/openneuro-app/src/scripts/users/user-query.tsx [hunk 10]
```diff
@@ -0,0 +1,81 @@
+// UserQuery.tsx
+
+import React from 'react';
+import { useParams } from 'react-router-dom';
+import { UserRoutes } from './user-routes';
+import FourOFourPage from '../errors/404page';
+
+// ORCID validation regex pattern
+const orcidPattern = /^\d{4}-\d{4}-\d{4}-\d{4}$/;
```
Comment (nellh) [inline packages/openneuro-app/src/scripts/users/user-query.tsx]: The ORCID can end in X as the last value is a checksum.

We should reuse this implementation from here:
https://github.com/OpenNeuroOrg/openneuro/blob/08e28d29991668fd032aabbeb1f2185d3b24238c/packages/openneuro-app/src/scripts/dataset/mutations/update-permissions.tsx#L10-L18

**Review by nellh — CHANGES_REQUESTED** [review]

File: packages/openneuro-app/src/scripts/users/__tests__/user-account-view.tsx (hunk 8 above)
Comment (thinknoack) [reply packages/openneuro-app/src/scripts/users/__tests__/user-account-view.tsx]: will update with query by orcid

File: packages/openneuro-app/src/scripts/users/__tests__/user-account-view.tsx
Comment (thinknoack) [reply packages/openneuro-app/src/scripts/users/__tests__/user-account-view.tsx:1]: updated

File: packages/openneuro-app/src/scripts/users/user-query.tsx (hunk 10 above)
Comment (thinknoack) [reply packages/openneuro-app/src/scripts/users/user-query.tsx]: added new validationUtils.ts that is now available to on-app

**Pull request merged**